hw1_mzw7af_vl4kz.py

```diff
@@ -57,6 +57,30 @@ def main():
     facts = {}
     rules = []
 
+    for line in sys.stdin:
+        argArray = line.strip().split(" ")
 
+        if argArray[0].lower() == "teach":
+            if argArray[3] == "=":
+                stringVar = ' '.join(argArray).split(" = ")[1]
+                teachVar(argArray[1], argArray[2], stringVar)
```

Comment:
isn't stringVar just argArray[4]? If it's an issue with split, you can specify how many times to split:

https://stackoverflow.com/questions/30636248/split-a-string-only-by-first-space-in-python